public/src/test/java/com/rackspace/salus/telemetry/api/web/client/TestMonitorControllerTest.java

```
@@ -114,4 +115,22 @@ public void testCreateTestMonitorAndEventTask_Success() throws Exception {
             .characterEncoding("utf-8"))
         .andExpect(status().isOk());
   }
+
+
+  @Test
+  public void testCreateTestMonitorAndEventTask_Failure() throws Exception {
+    String tenantId = RandomStringUtils.randomAlphabetic(8);
+    TestMonitorAndEventTaskRequest testMonitorAndEventTaskRequest = podamFactory
+        .manufacturePojo(TestMonitorAndEventTaskRequest.class);
+
+    when(testMonitorAndEventTaskService.getTestMonitorAndEventTask(anyString(), any()))
+        .thenThrow(new IllegalArgumentException("Unable to find matching metric name"));
+
+    mockMvc.perform(
+        MockMvcRequestBuilders.post("/v1.0/api/tenant/{tenantId}/test-monitor-event-task", tenantId)
+            .contentType(MediaType.APPLICATION_JSON)
+            .content(objectMapper.writeValueAsString(testMonitorAndEventTaskRequest))
+            .characterEncoding("utf-8"))
+        .andExpect(status().isBadRequest());
```

Note to self: I'll see why this unit test doesn't behave as expected here

---

yes. do let me know as well where changes were required.

---

Based on my comments above, I wasn't sure that enough validation was occurring to lead to an error like this.

I have a feeling the api clients hide the exceptions thrown.

---

with the changes suggested above, this expectation was finally validated as expected